<div style="text-align: right"> Dawid Demko, 254906 </div>

# *Comparative Analysis of Business Insider - Video Games Related Articles*
### In this short analysis two articles are presented, one of them focused on a top list of video games and the second one discussing one of the biggest video game events of the year.

   News websites tend to cast their net wide in order to gain the most readers possibly, if not for the content, at least for the possible ad revenue that comes with each viewing of the page. This is why some create articles about issues not related to the general idea behind the website. One of such websites is Business Insider - an American financial website posting articles related to video games.
   
   In this simple linguistic analysis two examples of such are shown. The first one concerns a list of top 50 video games, and the second one discusses E3, one of the biggest annual gaming events in the world. Both articles were collected using Beautiful Soup and then tokenized and tagged using NLTK. Below, there can be found both the process and the results.

### Article One: *Top 50 Video Games of All Time, According to Metacritic*

In [101]:
import requests
from bs4 import BeautifulSoup
website = requests.get("https://www.businessinsider.com/best-video-games-metacritic-2017-11?IR=T#48-bioshock-infinite-4")
soup = BeautifulSoup(website.content, 'html.parser')
links1 = soup.get_text()
# print(links1)

In [79]:
import nltk
from nltk.probability import FreqDist
from nltk.tokenize import RegexpTokenizer
nltk.download('stopwords')

[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\polac\AppData\Roaming\nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


True

In [100]:
stops = stopwords.words('english')
links1 = links1.lower()
tokenizer = RegexpTokenizer(r'\w+')
tokens1 = tokenizer.tokenize(links1)
tokens1 = [token for token in tokens1 if token not in stops]
# print(tokens1)

In [81]:
from nltk.tokenize import word_tokenize
toplist = word_tokenize(links1)
nltk.pos_tag(toplist)

[('top', 'JJ'),
 ('50', 'CD'),
 ('video', 'JJ'),
 ('games', 'NNS'),
 ('of', 'IN'),
 ('all', 'DT'),
 ('time', 'NN'),
 (',', ','),
 ('according', 'VBG'),
 ('to', 'TO'),
 ('metacritic', 'JJ'),
 ('insider', 'NN'),
 ('logo', 'VBD'),
 ('the', 'DT'),
 ('word', 'NN'),
 ('``', '``'),
 ('insider', 'NN'),
 ("''", "''"),
 ('.', '.'),
 ('set', 'VBN'),
 ('up', 'RP'),
 ('later', 'RBR'),
 ('close', 'RB'),
 ('icon', 'JJ'),
 ('two', 'CD'),
 ('crossed', 'JJ'),
 ('lines', 'NNS'),
 ('that', 'WDT'),
 ('form', 'VBP'),
 ('an', 'DT'),
 ("'", "''"),
 ('x', 'NN'),
 ("'", "''"),
 ('.', '.'),
 ('it', 'PRP'),
 ('indicates', 'VBZ'),
 ('a', 'DT'),
 ('way', 'NN'),
 ('to', 'TO'),
 ('close', 'VB'),
 ('an', 'DT'),
 ('interaction', 'NN'),
 (',', ','),
 ('or', 'CC'),
 ('dismiss', 'VB'),
 ('a', 'DT'),
 ('notification', 'NN'),
 ('.', '.'),
 ('menu', 'NN'),
 ('icon', 'VBZ'),
 ('a', 'DT'),
 ('vertical', 'JJ'),
 ('stack', 'NN'),
 ('of', 'IN'),
 ('three', 'CD'),
 ('evenly', 'RB'),
 ('spaced', 'VBD'),
 ('horizontal', 'JJ'),
 ('li

In [82]:
from collections import Counter
tags = nltk.pos_tag(tokens1)
count = Counter(tag for word, tag in tags)
print(count)

Counter({'NN': 1793, 'JJ': 925, 'NNS': 585, 'CD': 495, 'VBP': 292, 'VBD': 152, 'VBG': 147, 'RB': 140, 'VBZ': 105, 'VBN': 80, 'IN': 68, 'VB': 57, 'NNP': 21, 'JJS': 21, 'MD': 21, 'JJR': 10, 'PRP': 9, 'FW': 9, 'DT': 6, 'CC': 5, 'RP': 4, '$': 4, 'WDT': 2, 'RBR': 2, 'WP': 1, 'POS': 1, 'WP$': 1})


In the results of the first analysis, we can see that singular nouns (marked as *'NN'*) appear around two times as often as adjectives (*'JJ'*). Interestingly, plural nouns have a much lower rate of roughly 1 plural per 3 singular nouns. The article being a top list includes nearly as many cardinal digits (*'CD'*) as plural nouns, which is obvious, since plural nouns are most often preceeded by digits.

### Article Two: *E3 2021: Schedule, How to Watch, Times, What to Expect*

In [99]:
import requests
from bs4 import BeautifulSoup
website = requests.get("https://www.businessinsider.com/e3-2021-schedule-how-to-watch?IR=T")
soup = BeautifulSoup(website.content, 'html.parser')
links2 = soup.get_text()
# print(links2)

In [98]:
stops = stopwords.words('english')
links2 = links2.lower()
tokenizer = RegexpTokenizer(r'\w+')
tokens2 = tokenizer.tokenize(links2)
tokens2 = [token for token in tokens2 if token not in stops]
# print(tokens2)

In [88]:
from nltk.tokenize import word_tokenize
e3_article = word_tokenize(links2)
nltk.pos_tag(e3_article)

[('e3', 'JJ'),
 ('2021', 'CD'),
 (':', ':'),
 ('schedule', 'NN'),
 (',', ','),
 ('how', 'WRB'),
 ('to', 'TO'),
 ('watch', 'VB'),
 (',', ','),
 ('times', 'NNS'),
 (',', ','),
 ('what', 'WP'),
 ('to', 'TO'),
 ('expect', 'VB'),
 ('insider', 'NN'),
 ('logo', 'IN'),
 ('the', 'DT'),
 ('word', 'NN'),
 ('``', '``'),
 ('insider', 'NN'),
 ("''", "''"),
 ('.', '.'),
 ('set', 'VBN'),
 ('up', 'RP'),
 ('later', 'RBR'),
 ('close', 'RB'),
 ('icon', 'JJ'),
 ('two', 'CD'),
 ('crossed', 'JJ'),
 ('lines', 'NNS'),
 ('that', 'WDT'),
 ('form', 'VBP'),
 ('an', 'DT'),
 ("'", "''"),
 ('x', 'NN'),
 ("'", "''"),
 ('.', '.'),
 ('it', 'PRP'),
 ('indicates', 'VBZ'),
 ('a', 'DT'),
 ('way', 'NN'),
 ('to', 'TO'),
 ('close', 'VB'),
 ('an', 'DT'),
 ('interaction', 'NN'),
 (',', ','),
 ('or', 'CC'),
 ('dismiss', 'VB'),
 ('a', 'DT'),
 ('notification', 'NN'),
 ('.', '.'),
 ('menu', 'NN'),
 ('icon', 'VBZ'),
 ('a', 'DT'),
 ('vertical', 'JJ'),
 ('stack', 'NN'),
 ('of', 'IN'),
 ('three', 'CD'),
 ('evenly', 'RB'),
 ('spaced', 'V

In [93]:
from collections import Counter
tags = nltk.pos_tag(tokens2)
count = Counter(tag for word, tag in tags)
print(count)

Counter({'NN': 855, 'JJ': 316, 'NNS': 281, 'CD': 147, 'VBP': 101, 'VBD': 64, 'VBG': 62, 'RB': 56, 'VBZ': 46, 'VB': 31, 'VBN': 27, 'JJS': 26, 'NNP': 20, 'IN': 16, 'MD': 11, 'PRP': 7, 'DT': 6, 'FW': 4, 'RBR': 4, 'JJR': 3, '$': 3, 'CC': 3, 'RBS': 2, 'UH': 1})


In the analysis of the second article, the following can be said about the parts of speech that appear most often. *NNs* and *NNSs* appear in the 3 to 1 ratio. The ratio of *JJs* to *NNs* is similar and the ratio of cardinal digits to *NNs* is 1 to 5.81.

## Comparison of the Results

The following table presents the comparison of number of appearances of different parts of speech in both articles:


| article no. |  NN  | NNS |  JJ |  CD |
|:-----------:|:----:|:---:|:---:|:---:|
|     One     | 1793 | 585 | 925 | 495 |
|     Two     |  855 | 281 | 316 | 147 |

When it comes to nouns, both articles are similar with around 3 singular nouns for each plural. However, in article one statistically there appears one adjective for each two singular nouns, and in the second article the tendency is lower with nearly three singular nouns per each adjective. Moreover, the number of cardinal digits in the second article is also much lower with one digit for nearly six *NNs*, whereas in the first one cardinal digits appear more often - one for each 3.62 *NNs*. In both articles, a low number of verbs (*'VBP','VBD','VBG'*) can be noticed.

## Conclusion

This short comparative analysis showed that non-financial articles on *Business Insider* show a tendency to be written in a descriptive style focusing on nouns and their descriptors rather than actions or argumentation. This is caused by the nature of articles, both of which listed and explained features of either different video games or the event.